src/check/test/type_checking_integration.zig

```diff
@@ -1899,6 +1899,55 @@ test "top-level: type annotation followed by body should not create duplicate de
     }
 }
 
+// equirecursive static dispatch //
+
+test "check type - equirecursive static dispatch - motivating example (current behavior)" {
+    // This is the motivating example for equirecursive unification!
+    // Before RecursionVar was implemented, this would cause infinite loops during type checking
+    // because x.plus returns a numeric type that also needs .plus constraints, creating
+    // an infinite chain: ret.plus : ret, _ -> ret2, ret2.plus : ret2, _ -> ret3, ...
+    //
+    // With RecursionVar, we detect the recursive constraint and create a circular type
+    // that unifies equirecursively (structurally equal up to recursion point).
+    //
+    // NOTE: The .plus method is not yet implemented on numeric types, so this currently
+    // fails with TYPE DOES NOT HAVE METHODS. Once .plus is implemented, this test should
+    // pass and return a numeric type with preserved constraints.
+    const source = "(|x| x.plus(5))(7)";
```

Comment:
Can be done separately from this PR, but the example should not have any recursive types. If it does, that means that the dispatch constraints are being attached in the wrong place